### PR TITLE
created recipe-ingredient model, created tests to check join table, c…

### PR DESCRIPTION
**Pull request by ddavebbell**
…reated associations, tests pass

### CODE REVIEW

File: app/models/used_ingredient.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class UsedIngredient < ApplicationRecord
+	belongs_to :ingredients
```
Comment (somethinrother) [inline app/models/used_ingredient.rb]: Has the opposing has_many behaviour been added to these classes?

File: spec/models/used_ingredient_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  subject do
+    build(:used_ingredient)
+  end
+
+  describe 'when observing relationships' do
+    context 'with recipes' do
+      it 'belongs to an ingredient' do
+        ingredient = Ingredient.new(name: 'Test Ingredient')
+        recipe = Recipe.new(name: 'Test Recipe')
+        ingredient_mention = UsedIngredient.new(recipe_id: recipe.id, ingredient_id: ingredient.id)
```
Comment (somethinrother) [inline spec/models/used_ingredient_spec.rb]: used_ingredient instead of ingredient_mention

File: spec/models/used_ingredient_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        expect(ingredient_mention.recipe_id).to eq(recipe.id)
+      end
+    end
+    context 'with ingredients' do
+       it 'belongs to an recipe' do
+         ingredient = Ingredient.new(name: 'Test Ingredient')
+         recipe = Recipe.new(name: 'Test Recipe')
+         ingredient_mention = UsedIngredient.new(recipe_id: recipe.id, ingredient_id: ingredient.id)
```
Comment (somethinrother) [inline spec/models/used_ingredient_spec.rb]: Here as well

File: spec/models/recipe_spec.rb [hunk 4]
```diff
@@ -20,9 +20,24 @@
       end
     end
   end
+
+  describe  'observing recipes relationship' do
+    context 'with ingredients' do
+      it 'has ingredients' do
+      recipe = create(:recipe, name: subject.name)
```
Comment (somethinrother) [inline spec/models/recipe_spec.rb]: This should be indented

File: spec/models/recipe_spec.rb [hunk 5]
```diff
@@ -20,9 +20,24 @@
...
+      ingredient = Ingredient.new(name: 'Ingredient')
+      ingredient2 = Ingredient.new(name: 'Ingredient 2')
+      used_ingredient = UsedIngredient.new(recipe_id: recipe.id, ingredient_id: ingredient.id)
+      used_ingredient2 = UsedIngredient.new(recipe_id: recipe.id, ingredient_id: ingredient2.id)
+      expect(recipe).to be_valid
```
Comment (somethinrother) [inline spec/models/recipe_spec.rb]: I would hope for a more useful assertion here, something like expect(recipe.ingredients.length).to eq(2)

File: spec/factories.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 FactoryBot.define do
+
```
Comment (somethinrother) [inline spec/factories.rb]: delete this empty space plz

File: spec/models/ingredient_recipe.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+RSpec.describe IngredientsRecipes, type: :model do
+
+  subject do
+    build(:ingredient_recipe)
+  end
+
+  describe 'when observing relationships' do
+    context 'with recipes' do
+      it 'belongs to an ingredient' do
+        ingredient = Ingredient.new(name: 'Test Ingredient')
```
Comment (somethinrother) [inline spec/models/ingredient_recipe.rb]: We shouldn't be using Model.new here. Remember to use the rspec create and build syntax. This applies all over these tests

Comment (ddavebbell) [reply spec/models/ingredient_recipe.rb]: okay

**Pull request merged**